### PR TITLE
Require mapping from id to storage path (#374)

### LINKED ISSUE #374: Make it clear that it is expected to have a well defined bi-directional mapping between storage path and id
The current text in [4.3 Storage Hierarchies](https://ocfl.io/0.3/spec/#root-hierarchies) does not spell out that we expect there to be a well-defined bi-directional mapping between id and storage paths such that for an id the path is uniquely defined, and vice versa. 

See https://github.com/OCFL/spec/issues/369#issuecomment-531680861 and following discussion

### PR DESCRIPTION
**Pull request by zimeon**
Adds clarification, fixes #374

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -913,12 +913,12 @@ <h2>Storage Hierarchies</h2>
           hierarchies, the following restrictions apply:
         </p>
         <ol>
+          <li>There MUST be a well-defined mapping from an object identifier to a unique storage path</li>
```
Comment (awoods) [inline draft/spec/index.html]:
```
There MUST be a well-defined, [reproducible] mapping...
```

Comment (zimeon) [reply draft/spec/index.html]: Do you really want to add "reproducible"? Seems over the top to me

Comment (rosy1280) [reply draft/spec/index.html]: yeah, given feedback so far, i would agree that adding reproducible might be a bit much.

Comment (ahankinson) [reply draft/spec/index.html]: Perhaps "deterministic"?

Comment (zimeon) [reply draft/spec/index.html]: well-defined is deterministic ... see https://en.wikipedia.org/wiki/Well-defined

Comment (zimeon) [reply draft/spec/index.html]: agree with discussion to change from well-defined to deterministic

**Pull request merged**